src/SPC/doctor/item/LinuxMuslCheck.php

```diff
@@ -8,57 +8,122 @@
+                shell(true)->exec($prefix . $fix_path);
+            }
+            return true;
+        } catch (RuntimeException) {
+            return false;
+        }
     }
 
     /**
+     * @throws DownloaderException
      * @throws RuntimeException
-     * @noinspection PhpUnused
+     * @throws FileSystemException
      */
-    #[AsFixItem('fix-musl')]
-    public function fixMusl(array $distro): bool
+    public function installMuslWrapper(): void
     {
-        $rhel_install = 'wget https://musl.libc.org/releases/musl-1.2.4.tar.gz && tar -zxvf musl-1.2.4.tar.gz && \
-                         rm -f musl-1.2.4.tar.gz && cd musl-1.2.4 && 
-                         if [[ ! "$PATH" =~ (^|:)"/usr/local/musl/bin"(:|$) ]]; then echo "export PATH=/usr/local/musl/bin:$PATH" >> ~/.bash_profile
-                         fi && \
-                         ./configure --enable-wrapper=gcc && \
-                         make -j && make install && cd .. && rm -rf musl-1.2.4';
-        $install_cmd = match ($distro['dist']) {
-            'ubuntu', 'debian' => 'apt-get install musl musl-tools -y',
-            'alpine' => 'apk add musl musl-utils musl-dev',
-            'rhel' => $rhel_install,
-            'almalinux' => $rhel_install,
-            default => throw new RuntimeException('Current linux distro does not have an auto-install script for musl packages yet.'),
-        };
         $prefix = '';
         if (get_current_user() !== 'root') {
             $prefix = 'sudo ';
             logger()->warning('Current user is not root, using sudo for running command');
         }
-        try {
-            shell(true)->exec($prefix . $install_cmd);
-            return true;
-        } catch (RuntimeException) {
-            return false;
-        }
+        $musl_source = [
+            'type' => 'url',
+            'url' => 'https://musl.libc.org/releases/musl-1.2.4.tar.gz',
+        ];
+        Downloader::downloadSource('musl-1.2.4', $musl_source);
+        FileSystem::extractSource('musl-1.2.4', DOWNLOAD_PATH . '/musl-1.2.4.tar.gz');
+        $musl_gcc_install_cmd = 'cd ' . SOURCE_PATH . '/musl-1.2.4 && \
+                         ./configure --enable-wrapper=gcc && \
+                         make -j && make install';
+        shell(true)->exec($prefix . $musl_gcc_install_cmd);
+    }
+
+    /**
+     * @throws DownloaderException
+     * @throws RuntimeException
+     * @throws FileSystemException
+     */
+    public function installMuslCrossMake(): void
+    {
+        $arch = arch2gnu(php_uname('m'));
+        $musl_compile_source = [
+            'type' => 'url',
+            'url' => "https://musl.cc/{$arch}-linux-musl-cross.tgz",
+        ];
+        Downloader::downloadSource('musl-compile', $musl_compile_source);
+        FileSystem::extractSource('musl-compile', DOWNLOAD_PATH . "/{$arch}-linux-musl-cross.tgz");
```

Comment:
We should replace this with our own self-hosted pre-compiled version, like at https://github.com/DubbleClick/static-php-cli/releases/tag/tools but tagged on this repository or something.

For one, because musl.cc is not an official musl website, but second, because the version there is compiled with --enable-default-pie which means intl segfaults. This is not the case with the version I built myself.

Reply:
I can host these files on dl.static-php.dev. But I think it's better to build it directly in Actions and automatically upload to server, similar to [static-php-cli-hosted](https://github.com/crazywhalecc/static-php-cli-hosted).

Reply:
You can do that if you want, but I don't see much point in rebuilding it over and over. We would have to maintain patches for the repository ourselves because the main repo rarely gets updated and new gcc/linux headers/etc versions are not used by default.

This will very rarely update and mostly just be a big waste of time.

Reply:
I see, you'll have to manually edit the workflow to update the versions used, but I will post a step-by-step how to build them (on Alpine, RHEL may be overkill for an action runner) tomorrow.